### DISCUSSION TITLE
Tried to upgrade, and failed. ;(

### DISCUSSION THREAD

**Discussion by ViktorJp** [jasonacox/Powerwall-Dashboard#132] [Q&A] [answered]
Hi there... thanks for this amazing work and visualization!  I have a working version of your grafana dashboard, and noticed you updated it recently, so I figured I would try to upgrade.  I went through the upgrade process, and loaded the latest dashboard.json file, however, when I pull it up, I'm greeted by a blank screen with little spinning "waiting" indicators.  I can still pull up the old dashboard, and it continues to work.  I'm not sure what went wrong?  After importing the new dashboard.json, I'm pointing it to the same influx db and sun/moon db.  Could you please give me some suggestions to figure out why it's not making that connection?
![powerwall2](https://user-images.githubusercontent.com/97465574/202288165-0735e2e2-63fa-4d35-8195-ab940e4aca2f.jpg)

Thank you!
Viktor

**Comment by jasonacox** [comment, marked as answer]
Hi @ViktorJp  - how did you upgrade?  Did you run the `upgrade.sh` script?  The grafana display you are showing looks like the older 6.5.1 version.  I suspect grafana didn't update to 9.1.2. 

If you haven't already, do a full backup and attempt another upgrade: `./upgrade.sh`.

**Reply by ViktorJp** [reply]
Thanks for the suggestions, @jasonacox ... I ran the ./upgrade.sh command... Just ran it again, and it states that grafana is "up-to-date".  But you are right... I seem to be stuck on 6.5.1.  The only complaint during the update is about a python warning, but I've attached a screenshot below.  Is there a way to just force upgrade grafana?  Also, forgive my ignorance, but looking at the upgrade.sh file, I don't see any mention about a grafana download/install, unless it all somehow is already contained inside the docker file or elsewhere?
![upgrade](https://user-images.githubusercontent.com/97465574/202467729-d29987ff-c973-475f-8549-5926828b110f.jpg)

**Reply by jasonacox** [reply]
You are right!  The upgrade.sh script doesn't force the update, it is depending on docker-compose to see the version upgrade in the `powerwall.yml` definition file and upgrading it for us.  Can you check the grafana section of your `powerwall.yml` file?  It should show:

```yml
    grafana:
        image: grafana/grafana:9.1.2-ubuntu
```

It it does show the right version, you can force the upgrade this way:

```bash
docker stop grafana
docker rm grafana
docker images | grep grafana | awk '{print $3}' | xargs docker rmi -f
./compose-dash.sh up -d
```

**Reply by ViktorJp** [reply]
Thanks very much for those instructions - they will be good to have incase they're needed in the future.  Sure enough, my .yml file was still showing 6.5.1 for some reason.  That makes me concerned not knowing if other files aren't properly getting updated?  I ended up just making a change to that .yml file, and running the ./upgrade.sh again, and this time, I've got 9.1.2.  :)  I truly appreciate your help!!

**Comment by richardklam** [comment]
On a side note, I was struggling to update my Powerwall dashboard on my Synology Docker, and discovered it was failing to upgrade because I was using Telnet instead of SSH. Silly me!
